### ISSUE TITLE
Multitask object detection learning

### ISSUE THREAD

**Issue by dimimal** [open-mmlab/mmdetection#6790]
Hi,

I am using mmdetection library as part of my research. Im working on 2 different, but very similar tasks. I am wondering if its possible to use an instance segmentation model with 2 different heads (mask+object) and apply multi task learning on this. I guess what I have to do is to create a wrapper head class which takes the features from the Neck and add the individual heads there.

Has anyone done something similar before?

Thanks

**Comment by ZwwWayne** [comment]
Yes you can do that, you may see the example of [RoIHead for mask rcnn](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/roi_heads/standard_roi_head.py).

**Comment by AronLin** [comment]
> I am wondering if its possible to use an instance segmentation model with 2 different heads (mask+object) and apply multi task learning on this.

Of course yes. Solo uses bbox_head and mask_head to complete instance segmentation task, you can refer to it.

**Comment by dimimal** [comment]
@ZwwWayne How can I use 2 double heads (mask+object) together, and train the model with 2 different loss functions? The 2 tasks are 2 individual instance segmentation tasks. Do you think, by modifying the RoIHead task, can I achieve what I need?

**Comment by Iammuratc** [comment]
> @ZwwWayne How can I use 2 double heads (mask+object) together, and train the model with 2 different loss functions? The 2 tasks are 2 individual instance segmentation tasks. Do you think, by modifying the RoIHead task, can I achieve what I need?

Hi, I want to do something similar. Did you find a way?

**Comment by dimimal** [comment]
@Iammuratc  No, I had to drop this and take care of other tasks that appeared back then. I think you should combine the standardROIheads under a custom class and register it. But there is more, you have to change the data ingestion and the training process. Having the ability to freeze heads for one task over another when backpropagating for finetuning. I think its achievable with a bit of work and coffee :)

**Comment by Iammuratc** [comment]
@dimimal Thank you very much for your reply. I think I will give it a try to [this](https://github.com/SimonVandenhende/Multi-Task-Learning-PyTorch) first, then.
